spicedb: add feature overview (#16)
authzed/docs: docs/spicedb/feature-overview.md

[client libraries]: /reference/clients.md
[API Overview]: /reference/api.md

### HTTP API

Optionally, SpiceDB can also support handling JSON requests over HTTP.
This is accomplished by using the [gRPC Gateway], which translates requests into gRPC messages and forwards them to the port SpiceDB is configured to use for handling gRPC requests.
The API is documented via [OpenAPI], but may have breaking changes in the future.

For more information, see the [API Overview].

[gRPC Gateway]: https://github.com/grpc-ecosystem/grpc-gateway
[OpenAPI]: https://petstore.swagger.io/?url=https://raw.githubusercontent.com/authzed/authzed-go/main/proto/apidocs.swagger.json
[API Overview]: /reference/api.md

### Datastore

[Datastores][datastore] are the means of configuring the behavior and performance of persisting relationship and schema data.

To understand the trade-offs between various implementations of Datastores, see the [Selecting a Datastore][select-datastore] guide.

[datastore]: /reference/glossary.md#datastore
[select-datastore]: spicedb/selecting-a-datastore.md

#### Read-only

All datastores can be configured to operate in a read-only mode.

> When API requests that attempt to perform writes are made while in read-only, the API will return an [Unavailable status][grpc-status] with the reason `SERVICE_READ_ONLY`.

Reviewer: while in read-only mode

[dispatch]: /reference/glossary.md#dispatchers

### Dashboard

The dashboard serves an internal website for humans operating SpiceDB.
It currently includes simple instructions for how to configure [zed][zed] for the running instance.

[zed]: https://github.com/authzed/zed

## Integration Tests Entrypoint

The integration testing entrypoint runs a modified versions of API service intended for applications to use as a fixture for running tests.
Each [Bearer Token][bearer-token] provided by the client is allocated its own isolated, ephemeral datastore.
By generating a unique authentication token for each test, tests can be executed in parallel safely against a single instance of SpiceDB.

[bearer-token]: /reference/api.md#authentication

### GitHub Action

One of the best ways to consume the integration tests entrypoint is via the [SpiceDB GitHub Action][gha].

[gha]: https://github.com/authzed/action-spicedb

## Devtools Entrypoint

The devtools entrypoint is currently used to power the [Playground][playground].

> It has the capability to validate schemas with amounts of relationships and assertions that fit in memory.

Reviewer 2: ```suggestion
It has the capability to validate schemas with relationships and assertions, using an amount that fits in memory.
```

> It can also optionally persist and serve bundled schemas with their relationships and assertions such that they can be shared or imported via [zed][zed] or using flags designed to bootstrap SpiceDB.

Reviewer 2: ```suggestion
It can also optionally persist and serve schemas bundled with their relationships and assertions. These bundles can be shared or imported into SpiceDB via [zed][zed] or using flags for bootstrapping SpiceDB.
```

[playground]: https://play.authzed.com
[zed]: https://github.com/authzed/zed

### Language Server

[In the future][lsp-issue], this API will change to support the [Language Server Protocol][lsp], so that more tools than just the Playground are supported.

[lsp-issue]: https://github.com/authzed/spicedb/issues/179
[lsp]: https://microsoft.github.io/language-server-protocol/

## All Entrypoints

### Logging

Every entrypoint to SpiceDB supports configuring structured logging with various formats and log levels.

### Metrics

> Every entrypoint to SpiceDB has a configurable HTTP server that serves both a [Prometheus metrics endpoint][prom-endpoint] at `/metrics` and [Go runtime pprof][go-pprof] endpoints at `/debug/pprof`.

Reviewer 2: ```suggestion
Every entrypoint to SpiceDB has a configurable HTTP server that serves as both a [Prometheus metrics endpoint][prom-endpoint] at `/metrics` and [Go runtime pprof][go-pprof] endpoints at `/debug/pprof`.
```
